app/scripts/controllers/MainCtrl.js

```diff
@@ -1,12 +1,28 @@
 (function(){
-  function MainCtrl(Room, $scope){
+  function MainCtrl(Room, $uibModal, $scope){
     this.mainTitle = "/assets/images/bloc-logo-white.png";
     this.roomArray = Room;
     $scope.rooms = this.roomArray.all;
+
+    this.openModal = function(){
+      var modal = $uibModal.open({
+        templateUrl: '/templates/new_room.html',
+        controller: function($scope, $uibModalInstance){
```

Comment:
I'd prefer this to be in it's own file, but I think you were having issues getting that to work?

I really feel like `controller: "AddRoomCtrl as addRoom",` should work, and allow you to access the AddRoomCtrl in the `new_room.html` template by using `addRoom`. Maybe give that a quick try?